src/Microsoft.Health.Fhir.TemplateManagement.UnitTests/TemplateCollectionProviderFactoryTests.cs

```diff
@@ -96,38 +95,24 @@ public void GiveDefaultImageReference_WhenGetTemplateProviderWithEmptyToken_ADef
 
         public void GiveNewDefaultTemplateTarGzFile_WhenInitDefaultTemplate_DefaultTemplatesWillBeInit(string targzName, string imageReference, string templateFolder)
         {
-            CreateTarGz(targzName, templateFolder);
+            CompressTarGzFile(targzName, templateFolder);
             TemplateCollectionProviderFactory factory = new TemplateCollectionProviderFactory(_cache, Options.Create(_config));
             DefaultTemplateInfo templateInfo = new DefaultTemplateInfo(DataType.Hl7v2, imageReference, targzName);
             factory.InitDefaultTemplates(templateInfo);
             Assert.NotNull(factory.CreateProvider(imageReference, string.Empty));
         }
 
-        private void CreateTarGz(string outputTarFilename, string sourceDirectory)
+        private void CompressTarGzFile(string outputFile, string sourceFolder)
         {
-            using FileStream fs = new FileStream(outputTarFilename, FileMode.Create, FileAccess.Write, FileShare.None);
-            using Stream gzipStream = new GZipOutputStream(fs);
-            using TarArchive tarArchive = TarArchive.CreateOutputTarArchive(gzipStream);
-            AddDirectoryFilesToTar(tarArchive, sourceDirectory, true);
-        }
-
-        private void AddDirectoryFilesToTar(TarArchive tarArchive, string sourceDirectory, bool recurse)
-        {
-            if (recurse)
+            var artifacts = new Dictionary<string, byte[]>();
+            var files = Directory.EnumerateFiles(sourceFolder, "*.*", SearchOption.AllDirectories);
+            foreach (var file in files)
             {
-                string[] directories = Directory.GetDirectories(sourceDirectory);
-                foreach (string directory in directories)
-                {
-                    AddDirectoryFilesToTar(tarArchive, directory, recurse);
-                }
+                artifacts.Add(file, File.ReadAllBytes(file));
```

Comment:
Could be simplified to `var artifacts = files.ToDictionary(file => file, File.ReadAllBytes)`

Reply:
Thanks, fixed it.

Reply:
Thanks, fixed it.